src/server/serve.rs

```diff
@@ -207,8 +207,7 @@ where
 
         // Otherwise, return an error.
         _ = time::sleep(shutdown_timeout) => {
-            let message = "server exited before all connections were closed".to_owned();
-            Err(DynError::from(message))
+            Err("server exited before all connections were closed".into())
```

Comment:
Slight scope creep here but it's not a hot path, simple enough of a change, and doesn't warrant an additional commit message.